From: Alexander Martinz <amartinz@shiftphones.com>
Date: Fri, 19 Aug 2022 10:24:48 +0200
Subject: fixup! input: touchscreen: focaltech_fts: cleanup and support 5452
 chip

BTN_TOUCH reporting is required for at least Android userspace[1].

[1] - https://source.android.com/docs/core/input/touch-devices#touch-device-driver-requirements

Signed-off-by: Alexander Martinz <amartinz@shiftphones.com>
---
 drivers/input/touchscreen/focaltech_fts.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/input/touchscreen/focaltech_fts.c b/drivers/input/touchscreen/focaltech_fts.c
index 8b65848..b0e5295 100644
--- a/drivers/input/touchscreen/focaltech_fts.c
+++ b/drivers/input/touchscreen/focaltech_fts.c
@@ -161,7 +161,9 @@ static void fts_report_touch(struct fts_ts_data *data)
 			touchscreen_report_pos(data->input_dev, &data->prop, x, y, true);
 			input_report_abs(input_dev, ABS_MT_PRESSURE, z);
 			input_report_abs(input_dev, ABS_MT_TOUCH_MAJOR, maj);
+			input_report_key(data->input_dev, BTN_TOUCH, 1);
 		} else {
+			input_report_key(data->input_dev, BTN_TOUCH, 0);
 			input_mt_report_slot_inactive(input_dev);
 		}
 	}
